csharp-advanced-topics/VolatileInterlockedLockInCsharp/VolatileInterlockedLockInCsharp/Account.cs

```diff
@@ -7,6 +7,7 @@ public class Account
         private int _balanceLock;
         private int _balanceInterlocked;
         private object _lock = new object();
+        private readonly int _delay = new Random().Next(0, 100);
```

Comment:
Use Random.Shared here

Reply:
I think you are still misunderstanding my statement about the random delay. What I was trying to indicate is that we should compute a different amount of delay in each Task. That is this line here:
`Thread.Sleep(_delay);`
using my suggestion would become:
`Thread.Sleep(Random.Shared.Next(50,300);`
That way we have a little more "real world" wobble in the threads. Each one taking a different amount of time to execute.

Reply:
Understood. Will make the necessary change.